File: plugin/src/webOptimization.ts
import type { NodePath } from '@babel/core';
import { booleanLiteral, isIdentifier } from '@babel/types';
import type { CallExpression } from '@babel/types';

export function substituteWebCallExpression(path: NodePath<CallExpression>) {
  const callee = path.node.callee;
  if (isIdentifier(callee)) {
    const name = callee.name;
    if (name === 'isWeb' || name === 'shouldBeUseWeb') {
      path.replaceWith(booleanLiteral(true));
    }
  }
}

File: packages/react-native-worklets/plugin-oxc/src/transformer/web_optimization.rs
use oxc_ast::AstBuilder;
use oxc_ast::ast::Expression;
use oxc_span::SPAN;

/// Replaces `isWeb()` / `shouldBeUseWeb()` calls with `true`.
pub fn substitute_web_call_expression<'a>(node: &mut Expression<'a>, builder: AstBuilder<'a>) {
    let Expression::CallExpression(call) = node else {
        return;
    };

    let Expression::Identifier(callee) = &call.callee else {
        return;
    };

    let name = callee.name.as_str();
    if name == "isWeb" || name == "shouldBeUseWeb" {
        *node = builder.expression_boolean_literal(SPAN, true);
    }
}
